ADR/0017-posture-keeper.md

```diff
@@ -0,0 +1,59 @@
+# 17. Posture Keeper
+
+Date: 2023-02-18
+
+## Status
+
+Accepted
+
+## Context
+Posture keeper is a REST-based web service that allows clients to store and retrieve SAST analysis data for a specific git repository and commit SHA. The service is designed to be scalable, reliable, and secure, and uses MongoDB as its underlying data store. The service is implemented using Quarkus. By utilizing this service StoneSoup is going to be able to provides experiences that reflects the current state of security and/or quality posture of the applications.
+
+## Design Goals
+
+The following are the key design goals of the service:
+
+* Support storing and retrieving SAST analysis data for a specific git repository and commit SHA
+* Accept and produce JSON-formatted data
+* Use MongoDB as the underlying data store for efficient data retrieval and storage
+* Support secure authentication and authorization to ensure only authorized clients can upload and retrieve data
```

Comment:
You can likely make a much more specific statement here. Should the service utilize subject access reviews in order to conform with the general RBAC system shared by other stonesoup services?